src/reducers/players.js

```diff
@@ -83,6 +84,25 @@ const players = (state = initialState, action) => {
         },
       });
     }
+    case SHUFFLE_CARD: {
+      // formula from MDN => Math.floor(Math.random() * (max - min + 1)) + min ... here, min = 0
+      const newIndex = Math.floor(Math.random() * (state[action.playerId].deck.length + 1));
+      const deckCopy = state[action.playerId].deck.slice(0);
+      deckCopy.splice(newIndex, 0, action.cardId);
+
+      const handCopy = state[action.playerId].hand.slice(0);
+
+      if (handCopy.indexOf(action.cardId) !== -1) {
```

Comment:
There's a really article in the redux docs for common [immutable update patterns](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html). There is a section for inserting and removing items from arrays and I think we should use the same pattern here.